plutus-tx/test/laws/Main.hs

```diff
@@ -0,0 +1,582 @@
+                                   (negate <$> genIntegerPos, genIntegerPos)]
+      Ratio.unsafeRatio <$> genN <*> genD
+
+propAbsRound :: Property
+propAbsRound = property $ do
+  r <- forAllWith ppShow genRational
+  let rounded = Ratio.round r
+  let (n, _) = Ratio.properFraction r
+  assert (Plutus.abs rounded Plutus.>= Plutus.abs n)
+
+propRoundHalf :: Property
+propRoundHalf = property $ do
+  (n, f) <- forAllWith ppShow go
+  let r = Ratio.fromInteger n Plutus.+ f
+  let rounded = Ratio.round r
+  case (signum n, even n) of
+    (-1, False) -> rounded === n Plutus.- Plutus.one
+    (-1, True)  -> rounded === n
+    (0, _)      -> rounded === Plutus.zero
+    (1, False)  -> rounded === n Plutus.+ Plutus.one
+    _           -> rounded === n
+  where
+    go :: Gen (Integer, Plutus.Rational)
+    go = do
+      n <- genInteger
+      f <- case signum n of
+            (-1) -> pure . Ratio.negate $ Ratio.half
+            0    -> Gen.element [Ratio.half, Ratio.negate Ratio.half]
+            _    -> pure Ratio.half
+      pure (n, f)
+
+propRoundLow :: Property
+propRoundLow = property $ do
+  (n, f) <- forAllWith ppShow go
+  let r = Ratio.fromInteger n Plutus.+ f
+  let rounded = Ratio.round r
+  let truncated = Ratio.truncate r
+  rounded === truncated
+  where
+    go :: Gen (Integer, Plutus.Rational)
+    go = do
+      num <- Gen.integral . Range.constant 1 $ 135
+      let f = Ratio.unsafeRatio num 271
+      n <- genInteger
+      case signum n of
+        (-1) -> pure (n, Ratio.negate f)
+        0    -> (n,) <$> Gen.element [f, Ratio.negate f]
+        _    -> pure (n, f)
+
+propRoundHigh :: Property
+propRoundHigh = property $ do
+  (n, f) <- forAllWith ppShow go
+  let r = Ratio.fromInteger n Plutus.+ f
+  let rounded = Ratio.round r
+  let truncated = Ratio.truncate r
+  Plutus.abs rounded === Plutus.abs truncated Plutus.+ Plutus.one
+  where
+    go :: Gen (Integer, Plutus.Rational)
+    go = do
+      num <- Gen.integral . Range.constant 136 $ 270
+      let f = Ratio.unsafeRatio num 271
+      n <- genInteger
+      case signum n of
+        (-1) -> pure (n, Ratio.negate f)
+        0    -> (n,) <$> Gen.element [f, Ratio.negate f]
+        _    -> pure (n, f)
+
+-- Helpers
+
+instance Arg Plutus.Rational where
+  build = via separate (uncurry Ratio.unsafeRatio)
+
+varyRational :: CoGen Plutus.Rational
+varyRational = contramap separate vary
+
+separate :: Plutus.Rational -> (Plutus.Integer, Plutus.Integer)
+separate r = (Ratio.numerator r, Ratio.denominator r)
+
+genRational :: Gen Plutus.Rational
+genRational = do
+  num <- genInteger
+  den <- Gen.integral . Range.linearFrom 100 0 $ 200
+  pure . Ratio.unsafeRatio num $ den
+
+genInteger :: Gen Integer
+genInteger = Gen.integral . Range.linearFrom 0 (-100) $ 100
+
+genIntegerPos :: Gen Integer
+genIntegerPos = Gen.integral . Range.linearFrom 50 1 $ 100
+
+-- This is a hack to ensure coverage.
+--
```

Comment:
You could maybe use `{- ... -}` here, but no big deal.  Good comment though.

Reply:
I prefer `--`-style multilines, not least for all because they're easier to deal with in Neovim.